### PR TITLE
Rewrite milestone maintainer

### PR DESCRIPTION
**Pull request by mkorbi**
#### What type of PR is this:
/kind documentation

#### What this PR does / why we need it:
This PR is to open the discussion on https://github.com/kubernetes/sig-release/issues/1257 and adjust the overall positioning of the milestone maintainer. At the same time, a PR will be opened on kubernetes/community. 

The overall idea is to have a broader description of the milestone maintainer, this imho is located best the release description within the community repository, so we have a single source of truth. 
#### Which issue(s) this PR fixes:
https://github.com/kubernetes/sig-release/issues/1257

#### Special notes for your reviewer:
There will be another PR on the sig-release repository. Please first let's discuss at the 1257 issue to see the overall concept.

### CODE REVIEW

**Comment by mkorbi** [comment]
sorry I messed with the previous PR and closed it
/cc @LappleApple

File: release-team/README.md [hunk 1]
```diff
@@ -139,45 +139,15 @@ If you're interested in learning more about how the Release Team is selected, as
...
-- Assign an approver from the group you'll be maintaining issues/PRs for
-  i.e.,
-  - for SIGs: Request approval from a SIG Chair or Technical Lead
-  - for Release Team: Request approval from the current Release Team Lead or SIG Release Chairs
-  - for Special code reviewers: Request approval from SIG Release Chairs
-
-- Once your request is approved by the appropriate group, you can now assign to the PR to a SIG Release Chair for final approval.
-
-
+The actual process of maintaining milestones is lived by the SIGs and special elected contributors. The SIG release team steers the enhancment through the release cycle and take care that only well maintained enhancements will make it up to a new major release.
```
Comment (cpanato) [inline release-team/README.md:150]: s/enhancment/enhancement

**Comment by jeremyrickard** [comment]
@cpanato take a final look

File: release-team/README.md [hunk 2]
```diff
@@ -139,45 +139,15 @@ If you're interested in learning more about how the Release Team is selected, as
...
-- Assign an approver from the group you'll be maintaining issues/PRs for
-  i.e.,
-  - for SIGs: Request approval from a SIG Chair or Technical Lead
-  - for Release Team: Request approval from the current Release Team Lead or SIG Release Chairs
-  - for Special code reviewers: Request approval from SIG Release Chairs
-
-- Once your request is approved by the appropriate group, you can now assign to the PR to a SIG Release Chair for final approval.
-
-
+The actual process of maintaining milestones is lived by the SIGs and special elected contributors. The SIG release team steers the enhancement through the release cycle and take care that only well maintained enhancements will make it up to a new major release.
```
Comment (jeremyrickard) [inline release-team/README.md:150]: >The actual process of maintaining milestones is lived by the SIGs 

The actual process of maintaining milestones is owned by the SIGs

Comment (jeremyrickard) [reply release-team/README.md:150]: @mkorbi can you take this update this one and then we can merge it?

**Pull request merged**